interpreter/go/go.go

```diff
@@ -0,0 +1,164 @@
+		return nil, err
+	}
+
+	executablePath := C.CString(exec)
+	defer C.free(unsafe.Pointer(executablePath))
+
+	gd := &goData{}
+
+	//nolint:gocritic
+	status := C.symblib_goruntime_new(executablePath, &gd.goExecutable)
+	if status != C.SYMBLIB_OK {
+		return nil, fmt.Errorf("failed to create point resolver for '%s': %d",
+			C.GoString(executablePath), status)
+	}
+
+	return gd, nil
+}
+
+func (g *goData) Attach(_ interpreter.EbpfHandler, _ libpf.PID,
+	_ libpf.Address, _ remotememory.RemoteMemory) (interpreter.Instance, error) {
+
+	return &goInstance{
+		d: g,
+	}, nil
+}
+
+func (g *goInstance) GetAndResetMetrics() ([]metrics.Metric, error) {
+	return []metrics.Metric{
+		{
+			ID:    metrics.IDGoSymbolizationSuccess,
+			Value: metrics.MetricValue(g.successCount.Swap(0)),
+		},
+		{
+			ID:    metrics.IDGoSymbolizationFailure,
+			Value: metrics.MetricValue(g.failCount.Swap(0)),
+		},
+	}, nil
+}
+
+func (g *goInstance) Detach(_ interpreter.EbpfHandler, _ libpf.PID) error {
+	if g.d.goExecutable != nil {
+		C.symblib_goruntime_free(g.d.goExecutable)
+		g.d.goExecutable = nil
+	}
+	return nil
+}
+
+func (g *goInstance) Symbolize(symbolReporter reporter.SymbolReporter, frame *host.Frame,
+	trace *libpf.Trace) error {
+	if !frame.Type.IsInterpType(libpf.Native) {
+		return interpreter.ErrMismatchInterpreterType
+	}
+	sfCounter := successfailurecounter.New(&g.successCount, &g.failCount)
+	defer sfCounter.DefaultToFailure()
+
+	if g.d.goExecutable == nil {
+		return errors.New("point resolver is out of scope")
+	}
+
+	var symbols *C.SymblibSlice_SymblibResolvedSymbol
+	defer C.symblib_slice_symblibresolved_symbol_free(symbols)
+
+	//nolint:gocritic
+	status := C.symblib_point_resolver_symbols_for_pc(g.d.goExecutable,
+		C.uint64_t(frame.Lineno), &symbols)
+	if status != C.SYMBLIB_OK {
+		return fmt.Errorf("failed to do point lookup at 0x%x: %d",
+			frame.Lineno, status)
+	}
+
+	// Access resolved symbols
+	symbolsSlice := unsafe.Slice((*C.SymblibResolvedSymbol)(unsafe.Pointer(symbols.data)),
+		symbols.len)
+	if len(symbolsSlice) != 1 {
+		return fmt.Errorf("unexpected return for point lookup: %d", len(symbolsSlice))
+	}
+
+	lineNo := libpf.SourceLineno(symbolsSlice[0].line_number)
+	funcName := C.GoString(symbolsSlice[0].function_name)
+	sourceFile := C.GoString(symbolsSlice[0].file_name)
+
+	// The fnv hash Write() method calls cannot fail, so it's safe to ignore the errors.
+	h := fnv.New128a()
+	_, _ = h.Write([]byte(frame.File.StringNoQuotes()))
+	_, _ = h.Write([]byte(funcName))
+	_, _ = h.Write([]byte(sourceFile))
+	fileID, err := libpf.FileIDFromBytes(h.Sum(nil))
+	if err != nil {
+		return fmt.Errorf("failed to create a file ID: %v", err)
+	}
+
+	frameID := libpf.NewFrameID(fileID, libpf.AddressOrLineno(lineNo))
+
+	trace.AppendFrameID(libpf.GoFrame, frameID)
+
+	symbolReporter.FrameMetadata(&reporter.FrameMetadataArgs{
+		FrameID:      frameID,
+		FunctionName: funcName,
+		SourceFile:   sourceFile,
+		SourceLine:   lineNo,
+	})
```

Comment:
Now that the frame are inserted with synthesized `frameID`, nothing prevents doing this for all items of the slice and thus symbolizing the inlined functions. This is how hotspot interpreter works also. Since this is simple change of `if` to `for`, could this be done also now?

(I understand this was not possible in the original PR due to using the original FileID.)

Reply:
Inlined functions are currently not supported. Do you want the suggested change see now anyway or would it be fine to do it in a follow up once inlined functions are supported?

Reply:
Oh wow. I did not realize this is not supported in the rust point lookup! This was the primary reason why I was thinking to go this way. @christos68k Is there some reasons why the Go standard library `debug/gosym` which *appears* to be doing the same thing now is not a better choice?

Reply:
Ok, Seems the rust code also uses `mmap` among other things. It probably requires a bit more plumbing and fixing #352 among other things before getting the Go code to state where `debug/gosym` could be used. Perhaps we first merge this and revisit replacing this with `debug/gosym` after the other prerequisite work is done.

Reply:
> Now that the frame are inserted with synthesized frameID, nothing prevents doing this for all items of the slice and thus symbolizing the inlined functions. 

Added this with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/408/commits/ac927558e433c0e8a55f8f9942e5020097ff10d0